terraform/pkg/terraformBinary/binary.go

```diff
@@ -0,0 +1,85 @@
+package terraformbinary
+
+import (
+	"context"
+	"os"
+
+	version "github.com/hashicorp/go-version"
+	product "github.com/hashicorp/hc-install/product"
+	releases "github.com/hashicorp/hc-install/releases"
+	tfexec "github.com/hashicorp/terraform-exec/tfexec"
+	log "github.com/rs/zerolog/log"
+)
+
+func installTerraform(binary_dir string, terraform_version string) (string, error) {
+	log.Debug().Msgf("Trying to install terraform binary version %s", terraform_version)
+	installer := &releases.ExactVersion{
+		Product:    product.Terraform,
+		Version:    version.Must(version.NewVersion(terraform_version)),
+		InstallDir: binary_dir,
+	}
+
+	execPath, err := installer.Install(context.Background())
+	if err != nil {
+		return "", err
+	}
+
+	log.Debug().Msgf("Installed terraform binary to %s", execPath)
+
+	return execPath, nil
+}
+
+func ensureTerraform(binary_dir string, terraform_version string, terraform_dir string) (*tfexec.Terraform, error) {
+	log.Debug().Msgf("Ensuring terraform version %s", terraform_version)
+	execPath := binary_dir + "/" + "terraform"
```

Comment:
Use filepath.join() or something similar. This just seems prone to errors.
This applies to all file paths that are created through raw string manipulation

Reply:
Updated all paths to use proper path aggregation.